devel/ci/integration/tests/test_messages.py

```diff
@@ -0,0 +1,23 @@
+import time
+
+from conu import ConuException
+
+from .utils import get_sent_messages
+
+
+def test_push_composer_start(bodhi_container, db_container, rabbitmq_container):
+    try:
+        output = bodhi_container.execute(["bodhi-push", "--username", "ci", "-y"])
+    except ConuException as e:
+        assert False, str(e)
+    output = "".join(line.decode("utf-8") for line in output)
+    assert output.endswith("Sending composer.start message\n")
+    # Give some time for the message to go around
+    time.sleep(2)
```

Comment:
I wonder if we could use [asyncio.wait()](https://docs.python.org/3.6/library/asyncio-task.html#asyncio.wait) here so that we will wait *up to* a timeout but not always the length of the sleep here? This way we can react as the messages arrive, but also won't hang forever if they don't arrive. I once found a sleep statement in Bodhi that was causing its unit tests to take a good bit longer and got a pretty big boost by dropping it in d26d42a4468f07839ee649f4808c1ed45af7e382 ☺ What do you think?

Reply:
I'm not sure how to do that. What event will trigger the completion of the future in `wait()`? I don't think we can bubble up that event from the container to the host, where the tests are run. Can we?

Reply:
Yeah I didn't really consider the complexity of detecting changes from within the container.